framework/src/modules/http_api/controllers/node.js

```diff
@@ -250,37 +244,34 @@ NodeController.getPooledTransactions = function(context, next) {
 	// Remove filters with null values
 	filters = _.pickBy(filters, v => !(v === undefined || v === null));
 
-	return library.channel.invoke(`chain:${stateMap[state]}`, [
-		_.clone(filters),
-		(err, data) => {
-			if (err) {
-				return next(err);
-			}
-
-			const transactions = _.map(
-				_.cloneDeep(data.transactions),
-				transaction => {
-					transaction.senderId = transaction.senderId || '';
-					transaction.recipientId = transaction.recipientId || '';
-					transaction.recipientPublicKey = transaction.recipientPublicKey || '';
-
-					transaction.amount = transaction.amount.toString();
-					transaction.fee = transaction.fee.toString();
-
-					return transaction;
-				}
-			);
-
-			return next(null, {
-				data: transactions,
-				meta: {
-					offset: filters.offset,
-					limit: filters.limit,
-					count: parseInt(data.count),
-				},
-			});
-		},
-	]);
+	try {
+		const data = await library.channel.invoke('chain:getTransactionsFromPool', [
+			state,
+			_.clone(filters),
+		]);
+
+		const transactions = _.map(_.cloneDeep(data.transactions), transaction => {
```

Comment:
i would avoid using `_.map` but we can deal with it later because currently we have lots of places using them

Reply:
Why avoiding _.map? Just being curious :D

Reply:
Prefer native map